core_ml_modules/language_processing/lexicon.py

```diff
@@ -0,0 +1,160 @@
+class Lexicon:
+    """
+    Object containing lexical information on all words in the dataset
+    :ivar _all_words: a list of all words in the dataset as Word objects
+    :type _all_words: list(Word)
+    :ivar _features: dictionary of dictionaries where Words are sorted by linguistic features
+                     e.g. {
+                               "pos_tag": {
+                                   "noun": [Word("I"), Word("apples")],
+                                   "verb": [Word("eat")]
+                               },
+                               "original_str": {
+                                   "I": [Word("I")],
+                                   "eat": [Word("eat")],
+                                   "apples": [Word("apples")]
+                               }
+                           }
+    :type _features: dict[str, dict[str, list(str)]]
+    """
+
+    class _Word:
+        """
+        Word object containing its original form and messages which contain it, as well as any other defined linguistic features
+        :ivar _original_str: original form of the word
+        :type _original_str: str
+        :ivar _features: dictionary of the word's linguistic features and their values
+        :type _features: dict(str)
+        :ivar _messages: messages in which the word appears
+        :type _messages: list(list(str))
+        """
+        def __init__(self, original_str, features):
+            self._original_str = original_str
+            self._features = dict.fromkeys(features, None)
+            self._messages = []
+
+        def get_original_str(self):
+            return self._original_str
+
+        def get_feature_value(self, feature):
+            return self._features[feature]
+
+        def set_feature_value(self, feature, value):
+            self._features[feature] = value
+
+        def get_messages(self):
+            return self._messages
+
+        def add_message(self, message):
+            self._messages.append(message)
+
+    def __init__(self, messages, features):
```

Comment:
Please add a comment that this is the ctor for Lexicon, not for _Word. Nested classes are not much used in Python, so it's useful to be explicit here